File: ConsoleApp7/Account.cs
using System;

namespace Bankomat
{
    internal class Account
    {
        public string Number { get; private set; }
        public float Money { get; private set; } = 0;

        public Account()
        {
            Random random = new Random();

            for (int i = 0; i < 10; i++)
                Number += random.Next(10);
        }


        public void replenish(float amountMoney)
        {
            if (amountMoney <= 0)
                throw new InvalidOperationException("Ты кого наебать пытаешься");
            Money += amountMoney;
        }

        private bool CheckSolvency(float amountMoney)
        {
            if (amountMoney <= 0)
                return false;

            return Money >= amountMoney;
        }

        public void Pay(float amountMoney)
        {
            if (CheckSolvency(amountMoney))
            {
                Money -= amountMoney;
            }
            else
            {
                Console.WriteLine("Денег не хватает");
                Console.ReadKey();
            }

        }
    }
}

File: Bankomat/Account.cpp
#include "Account.h"


    Bankomat::Account::Account()
    {
        srand(time(nullptr));

        for (int i = 0; i < 10; i++)
            Number += std::to_string(rand() % 10);
    }

    std::string Bankomat::Account::GetNumber()
    {
        return Number;
    }

    float Bankomat::Account::GetMoney()
    {
        return Money;
    }

    void Bankomat::Account::replenish(float amountMoney)
    {
        if (amountMoney <= 0)
            throw std::exception("Ты кого наебать пытаешься");

        Money += amountMoney;
    }

    bool Bankomat::Account::CheckSolvency(float amountMoney)
    {
        if (amountMoney <= 0)
            return false;

        return Money >= amountMoney;
    }

    void Bankomat::Account::Pay(float amountMoney)
    {
        if (CheckSolvency(amountMoney))
        {
            Money -= amountMoney;
        }
        else
        {
            std::cout << "Денег не хватает";
            _getch();
        }
    }


